### PR TITLE
WT-6458 read row-store leaf pages with prepared updates in a single pass

### PR DESCRIPTION
**Pull request by keithbostic**
@kommiharibabu, here's the change I asked you about for your consideration. I think we can do this work in a single pass, but I'll defer to you on the final decision!

One change might not be obvious: I changed `WT_PAGE_ALLOC_AND_SWAP` to be a `__wt_calloc_def` call. Reading in a page must be single-threaded, and calling `WT_PAGE_ALLOC_AND_SWAP` implies other threads might be using these structures, and I found that confusing.

### CODE REVIEW

File: src/btree/bt_page.c [hunk 1]
```diff
@@ -584,72 +588,63 @@ __inmem_row_leaf(WT_SESSION_IMPL *session, WT_PAGE *page)
...
+            continue;
+
+        /* First prepared transaction setup. */
+        if (prepare == PREPARE_INSTANTIATE) {
+            WT_ERR(__wt_page_modify_init(session, page));
+            if (!F_ISSET(btree, WT_BTREE_READONLY))
+                __wt_page_modify_set(session, page);
+
+            /* Allocate the per-page update array. */
+            WT_ERR(__wt_calloc_def(session, page->entries, &page->modify->mod_row_update));
```
Comment (kommiharibabu) [inline src/btree/bt_page.c:604]: I think the memory allocation for `mod_row_update` is not considered as a page cache size. we can increase the `total_size` with that allocation memory.

Comment (keithbostic) [reply src/btree/bt_page.c:604]: Thanks, @kommiharibabu, fixed in 5ddec36c9.

**Comment by agorrod** [comment]
I think this is a great change, but I think we should hold off merging until after we have a stable release candidate.

**Comment by keithbostic** [comment]
This PR is ready to merge, but converting to a draft as it's blocked until after we have a stable release candidate.

**Comment by tammybailey** [comment]
evergreen retry

**Comment by tammybailey** [comment]
evergreen retry

**Pull request merged**